pcdsdevices/beam_stats.py

```diff
@@ -75,3 +84,98 @@ def __init__(self, prefix, *, name, skip_small_moves=True, atol=None,
             self.atol = atol
         super().__init__(prefix, name=name, skip_small_moves=skip_small_moves,
                          **kwargs)
+
+
+class Lcls(BaseInterface, Device):
+    '''
+    LCLS LINAC STATUS
+    bunch charge: 180 [pC]
+    repetition rate: 120 [Hz]
+    ebeam energy: 10.75 [GeV]
+    Vernier energy: -35 [MeV]
+    Photon Energy: 9.91 [keV]
+    BC2 peak current: 3201 [A]
+    electron bunch length: 23 [fs]
+    Last eLoss Scan: 0.93 [mJ]
+    '''
+    # Bunch Charge nC: FBCK:BCI0:1:CHRG	no desc
+    # Hard e-Energy GeV/c: BEND:DMPH:400:BDES - Desired B-Field
+    # Soft e-Energ GeV/c: BEND:DMPS:400:BDES - Desired B-Field
+    # Bunch Length fs: SIOC:SYS0:ML00:AO820 estimated FEL Pulse Duration (FWHM)
+
+    # BC2 IPK:	 FBCK:FB04:LG01:S5DES	BC2 Current set point
+
+    # Bunch charge (units: nC)
+    bunch_charge = Cpt(EpicsSignalRO, 'SIOC:SYS0:ML00:AO470', kind='normal')
+    # also have this one  FBCK:BCI0:1:CHRG (nC)
+    # estimated FEL Pulse Duration (FWHM) (units: fs)
+    bunch_length = Cpt(EpicsSignalRO, 'SIOC:SYS0:ML00:AO820', kind='normal')
+    # BLEN:LI21:265:WIDTH - not found
+    # Width of pulse (bunch length) Fsec
+
+    # Peak current after BC2 (units: A)
+    peak_current = Cpt(EpicsSignalRO, 'SIOC:SYS0:ML00:AO195', kind='normal')
+    # Last Eloss sxray energy (units: mJ)
+    eloss_energy = Cpt(EpicsSignalRO, 'PHYS:SYS0:1:ELOSSENERGY', kind='normal')
+    # Fast Feedback 6x6 Vernier (units: MeV)
+    vernier_energy = Cpt(EpicsSignalRO, 'FBCK:FB04:LG01:DL2VERNIER',
+                         kind='normal')
+
+    # FBCK Vernier SIOC:SYS0:ML00:CALC209 (units: MeV)
+    # DL2 Energy - FBCK:FB04:LG01:DL2VERNIER
+    # Vernier Limit (% of Bend Energy):	SIOC:SYS0:ML01:AO151 Vernier Scan Range
+    # Vernier Limit: SIOC:SYS0:ML01:CALC034 - Vernier Limit
+    # Vernie Crtrl w/limits: SIOC:SYS0:ML01:CALC033	- Vernier Ctrl w/ limits
+
+    # LCLSBEAM Event Rate (units: Hz)
+    beam_event_rate = Cpt(EpicsSignalRO, 'EVNT:SYS0:1:LCLSBEAMRATE',
+                          kind='normal')
+    # Photon eV HXR (units: eV)
+    photon_ev_hxr = Cpt(EpicsSignalRO, 'SIOC:SYS0:ML00:AO627', kind='normal')
+
+    # Abort Active Status
+    # [ 0] Disable [ 1] Enable
+    bykik_abort = Cpt(EpicsSignal, 'IOC:IN20:EV01:BYKIK_ABTACT', kind='normal')
+
+    # Abort Period in  beam shots
+    bykik_period = Cpt(EpicsSignal, 'IOC:IN20:EV01:BYKIK_ABTPRD',
+                       kind='normal')
+    # undulator k value - not the correct PV i think
+    undulator_k = Cpt(EpicsSignal, 'USEG:UND1:150:KACT', kind='normal')
+
+    def bykik_status(self):
+        """
+        Returns status of bykik abort
+        """
+        status = self.bykik_abort.get()
```

Comment:
```suggestion
        status = self.bykik_abort.get(as_string=True)
```

Ref:
```
In [1]: import ophyd

In [2]: sig = ophyd.EpicsSignalRO('IOC:IN20:EV01:BYKIK_ABTACT', name='sig')

In [3]: sig.get()
Out[3]: 1

In [4]: sig.get(as_string=True)
Out[4]: 'Enable'
```

Reply:
This is great, thank you. Should I also be doing something like this:
```
    def bykik_disable(self):
        """
        Disables bykik abort
        """
        return self.bykik_abort.put(value='Disable')

    def bykik_enable(self):
        """
        Enables bykik abort
        """
        return self.bykik_abort.put(value='Enable')
```
instead of using `0` and `1` respectively?

Reply:
That should work, sure. 

Additionally, if we always want to deal with these in terms of their string equivalents, it's probably appropriate to make the `EpicsSignal` component with `string=True` to begin with (i.e., `Cpt(EpicsSignal, ..., string=True)`). Give that a try and see what just `.get()` returns.

Reply:
```
In [1]: import ophyd

In [2]: sig = ophyd.EpicsSignalRO('IOC:IN20:EV01:BYKIK_ABTACT', name='sig', string=True)

In [3]: sig.get()
Out[3]: 'Disable'
```
😊